app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.test.ts

```diff
@@ -0,0 +1,191 @@
+import { ControllerMessenger } from '@metamask/base-controller';
+import {
+  MetaMetricsDataDeletionController,
+  type MetaMetricsDataDeletionControllerMessengerActions,
+} from './metametrics-data-deletion';
+
+describe('MetaMetricsDataDeletionController', () => {
+  describe('createMetaMetricsDataDeletionTask', () => {
+    it('creates a data deletion task and stores ID when user is participating in metrics tracking', async () => {
+      const mockMetaMetricsId = 'mockId';
+      const mockTaskId = 'mockTaskId';
+      const { controller, dataDeletionService } = setupController({
+        options: {
+          getMetaMetricsId: jest.fn().mockReturnValue(mockMetaMetricsId),
+          getParticipateInMetrics: jest.fn().mockReturnValue(true),
+        },
+        dataDeletionService: {
+          createDataDeletionRegulationTask: jest
+            .fn()
+            .mockResolvedValue(mockTaskId),
+          fetchDeletionRegulationStatus: jest.fn().mockResolvedValue('UNKNOWN'),
+        },
+      });
+
+      await controller.createMetaMetricsDataDeletionTask();
+      expect(
+        dataDeletionService.createDataDeletionRegulationTask,
+      ).toHaveBeenCalledWith(mockMetaMetricsId);
+      expect(
+        dataDeletionService.createDataDeletionRegulationTask,
+      ).toHaveBeenCalledTimes(1);
+      expect(
+        dataDeletionService.fetchDeletionRegulationStatus,
+      ).toHaveBeenCalledTimes(1);
+      expect(controller.state).toMatchObject({
+        metaMetricsDataDeletionId: mockTaskId,
+        metaMetricsDataDeletionDate: expect.any(Number),
+        metaMetricsDataDeletionStatus: 'UNKNOWN',
+        participateInMetricsDuringDeletion: true,
+      });
```

Comment:
Are we sure we want to use `toMatchObject` instead of `toStrictEqual` throughout the tests in this file?

`toMatchObject` matches a subset of the object's properties, so the tests would pass even if `controller.state` were e.g. `{ metaMetricsDataDeletionId: mockTaskId }`.

```suggestion
      expect(controller.state).toStrictEqual({
        metaMetricsDataDeletionId: mockTaskId,
        metaMetricsDataDeletionDate: expect.any(Number),
        metaMetricsDataDeletionStatus: 'UNKNOWN',
        participateInMetricsDuringDeletion: true,
      });
```